Source/FakeItEasy/Core/FakeManager.AutoFakePropertyRule.cs

```diff
@@ -11,7 +11,7 @@ public partial class FakeManager
         private class AutoFakePropertyRule
             : IFakeObjectCallRule
         {
-            public FakeManager FakeManager { get; set; }
+            public FakeManager FakeManager { private get; set; }

```

Comment:
Agreed, but I think we can deal with this in a separate PR. This change only highlights the design problem by minimising the accessibility to what is needed, and thus highlights the design problem.